db/changes.go

```diff
@@ -250,8 +250,19 @@ func (db *DatabaseCollectionWithUser) buildRevokedFeed(ctx context.Context, ch c
 				// If its less than we can send a standard revocation with Sequence ID as above.
 				// Otherwise: we need to determine whether a previous revision of the document was in the channel prior
 				// to the since value, and only send a revocation if that was the case
+
 				if logEntry.Sequence > sinceVal {
-					requiresRevocation, err := db.wasDocInChannelPriorToRevocation(ctx, logEntry.DocID, singleChannelCache.ChannelID().Name, revocationSinceSeq)
+					// Get doc sync data so we can verify the docs grant history
+					syncData, err := db.GetDocSyncData(ctx, logEntry.DocID)
+					if err != nil {
+						base.InfofCtx(ctx, base.KeyChanges, "Error getting sync data, seq: %v in channel %s, skipping document. Error: %v", seqID, base.UD(singleChannelCache.ChannelID()), err)
+						continue
+					}
+					if syncData.History == nil {
+						base.TracefCtx(ctx, base.KeyChanges, "Document history empty: %s, no revocation sent", base.UD(logEntry.DocID))
+						continue
+					}
```

Comment:
Last suggestion: I think we can drop this check - we don't write `syncData` with nil history, but even if we did, all of the following code should work as-is using only `ChannelSet`/`ChannelSetHistory`

```suggestion
```

Reply:
Addressed in https://github.com/couchbase/sync_gateway/pull/5930/commits/347eb319699a001848ede4f74b5231e1e7f3fd36